app/models/user.rb

```diff
@@ -0,0 +1,22 @@
+class User < ApplicationRecord
+  has_many :votes
+  has_many :works, through: :votes
+
+  def already_voted?(work)
+    return self.works.include?(work)
+  end
```

Comment:
I like that you've made this a model method.

If you wanted to make the interface even simpler / more foolproof, you could define it as a validation on `Vote`, so that it's impossible to create a duplicate vote. Look up "rails validation uniqueness scope"